helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -295,7 +296,7 @@ public void close() {
   // Coordinate baseline recalculation and partial rebalance according to the cluster changes.
```

Comment:
When computeNewIdealStates returns, do we need to do a putAll? I think we could just use the same reference for newIdealStates?

Reply:
Yes, we can. But the code would be a little bit awkward. Note there is a try-catch block. If we use the returned map directly, I need to set the to be null. Then if the wagedrebalance calculation fails, I set it with an empty map within the catch block. So the following non-waged rebalance logic can keep adding to that map.
I feel it would be reader unfriendly and potentially harder to maintain.